src/main.rs

```diff
@@ -29,6 +29,17 @@ fn notmain() -> Result<i32> {
     // NOTE `skip` the name/path of the binary (first argument)
     let raw_args = env::args().skip(1).collect::<Vec<_>>();
 
+    // If there's no argument provided, print the help message
+    if matches!(a.as_slice(), [] | ["--help" | "-h"]) {
```

Comment:
```suggestion
    if matches!(raw_args.as_slice(), [] | ["--help" | "-h"]) {
```

Reply:
Ahh, it seems not working: 
<img width="720" alt="image" src="https://github.com/user-attachments/assets/8fc283e4-f8c9-4be7-9d0f-49f731510bd5">

I tried to make Vec<String> to Vec<&str> in `matches!`, but it is somehow longer after formatting:

```
  if matches!(
      raw_args
          .iter()
          .map(|s| s.as_str())
          .collect::<Vec<&str>>()
          .as_slice(),
      [] | ["--help"] | ["-h"]
  ) {
      ...

```

Reply:
Ah dang, I didn't consider that when I tried it locally